apollo-router/src/spec/schema.rs

```diff
@@ -68,20 +67,28 @@ impl Schema {
         let recursion_limit = ast.recursion_limit();
         tracing::trace!(?recursion_limit, "recursion limit data");
 
-        // TODO: run full compiler-based validation instead?
-        let errors = ast.errors().cloned().collect::<Vec<_>>();
-        if !errors.is_empty() {
-            let errors = ParseErrors {
-                raw_schema: schema.to_string(),
-                errors,
+        let diagnostics = compiler
+            .validate()
+            .into_iter()
+            .filter(|err| err.data.is_error())
+            .collect::<Vec<_>>();
+
+        // Only bail out on parser errors for now: validation errors will be checked with the query
+        // planner result
```

Comment:
If no request can ever go through because of a (validation) error in the schema, I think it would be better to bail here during startup.

Reply:
Ya, it depends a bit on if we want to go all-in on apollo-rs validation immediately or test it on (our own?) prod traffic first. I've done this conditionally with the latter case in mind.
That said i'm not that worried about schema validation having false positives at the moment, and false negatives are not a big problem.